src/core/Akka/Util/ByteString.cs

```diff
@@ -544,27 +450,17 @@ public int CopyTo(byte[] buffer, int index, int count)
             if (index < 0 || index >= buffer.Length) throw new ArgumentOutOfRangeException(nameof(index), "Provided index is outside the bounds of the buffer to copy to.");
             if (count > buffer.Length - index) throw new ArgumentException("Provided number of bytes to copy won't fit into provided buffer", nameof(count));
 
-            count = Math.Min(count, _count);
-            var remaining = count;
-            var position = index;
-            foreach (var b in _buffers)
-            {
-                var toCopy = Math.Min(b.Count, remaining);
-                Array.Copy(b.Array, b.Offset, buffer, position, toCopy);
-                position += toCopy;
-                remaining -= toCopy;
-
-                if (remaining == 0) return count;
-            }
-
-            return 0;
+            count = Math.Min(count, Count);
+            _memory.Span.CopyTo(buffer.AsSpan(index, count));
```

Comment:
NIT: IMO at bare minimum this could/should call into `public int CopyTo(ref Memory<byte> buffer, int index, int count)`. But in general the `CopyTo` overloads can be consolidated.


Additional NIT: Torn as to whether the smaller IL is a benefit compared to lacking bounds checks. (OTOH, this is 'I'd have to sharplab and benchmark to reliably comment.)